TestApp/Source/Android/app/CMakeLists.txt

```diff
@@ -71,6 +69,12 @@ target_link_libraries(BabylonNativeJNI
         ${CMAKE_CURRENT_LIST_DIR}/../../../../build/Library/Dependencies/bgfx.cmake/libastc-codec.a
         ${CMAKE_CURRENT_LIST_DIR}/../../../../build/Library/Dependencies/SPIRV-Cross/libspirv-cross-core.a
         ${CMAKE_CURRENT_LIST_DIR}/../../../../build/Library/Dependencies/SPIRV-Cross/libspirv-cross-glsl.a
+        ${CMAKE_CURRENT_LIST_DIR}/../../../../build/Library/Dependencies/SPIRV-Cross/libspirv-cross-hlsl.a
```

Comment:
This whole block doesn't appear in the CMakeLists.txt file for the Windows targets. Maybe this was already discussed in a prior PR -- if so, sorry for asking redundant questions -- but doesn't CMake do the linking for you so long as you specify the hierarchical relationship?

Reply:
Also, does the Android TestApp need to be in its own separate CMake file? Win32 and UWP are both built from the same one, and I think our hope was to have all the platforms share as much as possible as an example to consumers trying to build cross-plat apps. Is there a way we could get Android to share more with the other platforms?

Reply:
Not that CMake part because it's handled by gradle. It's very specific to Android. Gradle handles running cmake, making the apk, signing ,... Maybe we can change the build to only copy the .so. Not sure there is much value in doing that.

Reply:
Okay, I think there's definitely something about this build system we're not understanding the same way. This CMakeLists.txt file never calls add_library on the Babylon Native libraries; and, looking through the broader changes, I see no path for the top-level CMakeLists.txt to know that _this_ CMakeLists.txt exists at all. It looks like these two things are attempting to be two completely separate build systems, one for the library and one for the app, instead of a single top-level build system designed to make use of the [Android CMake toolchain](https://developer.android.com/ndk/guides/cmake#the_cmake_toolchain_file) to generate the build, sharing as much as possible with the TestApps from other platforms and including as few Android-specific resources as possible. I haven't dived into this deeply enough to fully understand what kind of platform-specific changes might be necessary, but I would not expect that our only alternative would be to run two separate builds and link all the shared libraries by hand (at which point I'm not sure there's any purpose in using CMake anymore). Does the Android CMake toolchain not give us some other options for how to do this?

Reply:
That's right! 2 build systems. 1 CMake to build the library (with android tool chain cmake) and 1 cmake called by gradle for the .cpp in the android studio project. I've added in this PR a second Android project that can be compiled entirely with cmake, without Android studio. it might be possible with some cmake install command to make the apk and run some adb command to push the binary on the device (simulator of real device). But for the project with java activity, I would not try :)
Now, the inverse might also be possible. That the gradle build running that cmake also builds the library (maybe you are refering to that mecanism). That would look more like VS solution building the library and the testapp.